modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.server/src/main/scala/edu/gemini/seqexec/web/server/common/LogInitialization.scala

```diff
@@ -6,73 +6,77 @@ import java.util.logging.LogManager
 import scala.io.{Codec, Source}
 import scalaz._
 import Scalaz._
-import scalaz.effect.IO
+import scalaz.concurrent.Task
 
-trait LogInitialization {
-  sealed trait LogConfError
-  case class ConfigurationNotFound(s: String) extends LogConfError
-  case class LogDirNotFound(f: File) extends LogConfError
-  case object ConfigurationLoadingError extends LogConfError
+trait AppBaseDir {
+  /**
+    * Calculates the base dir of the application based on the location of "this" class jar file
+    * It will throw an exception if unable to find the base dir
+    */
+  def baseDir: Task[File] = Task.delay {
+    val clazz = this.getClass
+    val fileName = clazz.getResource(s"/${clazz.getName.replace(".", System.getProperty("file.separator"))}.class").getFile
 
-  type LogConf[A] = EitherT[IO, LogConfError, A]
```

Comment:
Exceptions in `Task` are analogous to those in `IO`. You can make exceptions observable via `.attempt` which takes `Task[A]` to `Task[Throwable \/ A]` or via combinators based on that operation. If the task is run via an `.unsafeXXX` method that doesn't trap exceptions then any such exception will be thrown on the calling thread. Exceptions on other thread pools are routed back to the caller. There is no notion of "detached" asynchronous execution; every executing task has a logical "caller".